File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        L = 0
        vowel_count = 0
        res = float("-inf")
        for R in range(len(s)):
            vowel_count += 1 if s[R] in vowels else 0
            if R - L + 1 > k:
                vowel_count -= 1 if s[L] in vowels else 0
                L += 1
            res = max(res, vowel_count)
        return res
    
# abciiidef
# vowel count = 2

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
        int curr = 0, res = 0;
        int n = s.size();
        for (int i = 0, j = 0; j < n; ++j){
            if (vowels.find(s[j]) != vowels.end()) {
                curr++;
            }

            while (j - i + 1 > k) {
                if (vowels.find(s[i]) != vowels.end()) {
                    curr--;
                }
                ++i;
            }

            res = max(res, curr);
        }
        return res;
    };
};